src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -281,8 +281,27 @@ private async Task<FunctionMetadataResponse> GetFunctionMetadataAsync(string fun
 
                 foreach (var func in functionMetadataList)
                 {
-                    response.FunctionMetadataResults.Add(func);
+                    if (func.RawBindings is null)
+                    {
+                        throw new InvalidOperationException($"Functions must declare at least one binding. No bindings were found in the function ${nameof(func)}.");
+                    }
+
+                    if (func is null)
```

Comment:
This should go above the previous if statement - you will null ref before reaching here otherwise.